logic/src/commonMain/kotlin/com/wire/kalium/logic/data/e2ei/E2EIRepository.kt

```diff
@@ -301,29 +304,37 @@ class E2EIRepositoryImpl(
         Either.Right(e2EIClient.getOAuthRefreshToken())
     }
 
-    override suspend fun fetchFederationCertificates(): Either<CoreFailure, Unit> = userConfigRepository.getE2EISettings().flatMap {
-        wrapApiRequest {
-            acmeApi.getACMEFederation(Url(it.discoverUrl).protocolWithAuthority)
-        }.flatMap { data ->
-            mlsClientProvider.getMLSClient().flatMap { mlsClient ->
-                wrapMLSRequest {
-                    mlsClient.registerIntermediateCa(data)
+    override suspend fun fetchFederationCertificates(): Either<CoreFailure, Unit> = discoveryUrl()
+        .flatMap {
+            wrapApiRequest {
+                acmeApi.getACMEFederation(Url(it).protocolWithAuthority)
+            }.flatMap { data ->
+                mlsClientProvider.getMLSClient().flatMap { mlsClient ->
+                    wrapMLSRequest {
+                        mlsClient.registerIntermediateCa(data)
+                    }
                 }
             }
         }
-    }
+
+    override fun discoveryUrl(): Either<CoreFailure, String> =
```

Comment:
the idea of this function is to have a central place where the errors with e2ei url are caught